pkg/cloud/azure/interfaces.go

```diff
@@ -30,9 +28,17 @@ type Spec interface {
 }
 
 // Service is a generic interface used by components offering a type of service.
-// example: Network service would offer get/createorupdate/delete.
+// Example: virtualnetworks service would offer Reconcile/Delete methods.
 type Service interface {
+	Reconcile(ctx context.Context, spec Spec) error
+	Delete(ctx context.Context, spec Spec) error
+}
+
+// GetterService is a temporary interface used by components which still require Get methods.
+// Once all components move to storing provider information within the relevant
+// Cluster/Machine specs, this interface should be removed.
+type GetterService interface {
 	Get(ctx context.Context, spec Spec) (interface{}, error)
-	CreateOrUpdate(ctx context.Context, spec Spec) error
```

Comment:
Please do not change these interfaces, a Service always implements Get/CreateOrUpdate/Delete. A reconciler on other hand implements reconcile and delete.

We would want to define reconcilers in a common location if its reused in multiple places which isnt the case and only cluster actuator uses them. Please move the reconcile logic to the actuator themselves

Reply:
See my comments below.